float32 x
float32 y
float32 v_x
float32 v_y
float32 angle
float32 angular_velocity
